### PR TITLE
planner: new cost formula for IndexJoin

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #35240

Problem Summary: planner: new cost formula for IndexJoin

### What is changed and how it works?

On modelver2, consider double read cost for IndexJoins,  which is `numDoubleReadTasks * seekFactor / concurrency`

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: planner/core/plan_cost.go [hunk 1]
```diff
@@ -498,7 +506,23 @@ func (p *PhysicalIndexJoin) GetCost(outerCnt, innerCnt float64, outerCost, inner
...
+	}
+	// estimate the double read cost for IndexJoin: (double-read-tasks * seek-factor) / concurrency
+	seekFactor := p.ctx.GetSessionVars().GetSeekFactor(nil)
+	batchSize := math.Max(1.0, float64(p.ctx.GetSessionVars().IndexJoinBatchSize))
+	concurrency := math.Max(1.0, float64(p.ctx.GetSessionVars().IndexLookupJoinConcurrency()))
+	// distRatio indicates how many requests corresponding to a batch, current value is from experiments.
+	// TODO: estimate it by using index correlation or make it configurable.
+	distRatio := 40.0
+	numDoubleReadTasks := (doubleReadRows / batchSize) * distRatio
+	return (numDoubleReadTasks * seekFactor) / concurrency
```
Comment (AilinKid) [inline planner/core/plan_cost.go:525]: do we need to adjust any test cases？

**Review by AilinKid — COMMENTED** [review]
Rest LGTM

**Review by winoros — APPROVED** [review]

File: planner/core/plan_cost.go (hunk 1 above)
Comment (qw4990) [reply planner/core/plan_cost.go:525]: I'll add some dedicated test cases after finishing cost model ver2.

**Review by AilinKid — APPROVED** [review]

**Pull request merged**